otter/util/pure_http.py

```diff
@@ -27,18 +28,20 @@ class Request(object):
 
     treq = logging_treq
 
-    @inlineCallbacks
-    def perform_effect(self, dispatcher):
-        """
-        Perform the request with treq.
 
-        :return: A two-tuple of (HTTP Response, content as bytes)
-        """
-        response = yield self.treq.request(self.method.upper(), self.url,
-                                           headers=self.headers,
-                                           data=self.data, log=self.log)
-        content = yield self.treq.content(response)
-        returnValue((response, content))
+@deferred_performer
+@inlineCallbacks
+def perform_request(dispatcher, intent):
+    """
+    Perform the request with treq.
+
+    :return: A two-tuple of (HTTP Response, content as bytes)
+    """
+    response = yield intent.treq.request(intent.method.upper(), intent.url,
```

Comment:
Now that there is clear decoupling of intent and performer, should implementation detail like `treq` be there in `intent`?